src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py

```diff
@@ -774,12 +776,22 @@ def insert_into_statement(
     return f"{INSERT}{INTO}{table_name}{table_columns}{project_statement([], child)}"
 
 
-def batch_insert_into_statement(table_name: str, column_names: List[str]) -> str:
+def batch_insert_into_statement(
+    table_name: str, column_names: List[str], paramstyle: str
+) -> str:
+    num_cols = len(column_names)
+    if paramstyle == "qmark":
+        placeholder_marks = [QUESTION_MARK] * num_cols
+    elif paramstyle == "numeric":
+        placeholder_marks = [f"{SINGLE_COLON}{i+1}" for i in range(num_cols)]
+    else:  # for pyformat and format
+        placeholder_marks = [PERCENT_S] * num_cols
```

Comment:
It seems like we previously didn't support format/pyformat/numeric binding at all.. should this be considered a new feature?

Reply:
sure. I don't mind